### ISSUE TITLE
Desenvolvimento Back-end

### ISSUE THREAD

**Issue by nicholasio** [WordPressBeloHorizonte/horizon-theme#70]
Discussões acerca sobre o desenvolvimento back-end, o que usar, o que não usar e como usar.

**Comment by rodrigo-brito** [comment]
Não vai ficar muita configuração para o Customizr não? As configurações de redes sociais e mapas podiam ser pelo Theme Options.

**Comment by leobaiano** [comment]
Como vamos fazer os CPTs e Meta box com custom fields necessários? Penso que podemos copiar as classes de cpt, metabox e taxonomy do odin, o que acham?

Sobre as configurações como marca, redes sociais e mapa pode ser com nas opções do tema mesmo.

**Comment by rodrigo-brito** [comment]
O Odin também tem uma classe bacana para as opções de tema.

**Comment by rodrigo-brito** [comment]
Os textos dos blocos vãos ser gerenciados por onde?

**Comment by leobaiano** [comment]
Sim, tem mesmo @rodrigo-brito, poderíamos copiar. Agora lembro que no inicio do projeto alguém falou sobre retirar o Odin para usar alguma outra coisa que não lembro agora.

**Comment by brenoalvs** [comment]
Concordo com o @leobaiano sobre o odin, acho que não precisamos de mais do que isso.

Essas configurações "avançadas" pra serem feitas no customizer exigiriam que a gente criasse novas classes para os controles.

Os textos dos blocos podem ser na theme options também.
Mas já vi um tema que permitia criar páginas e usá-las como seções na home, só não sei como seria fazer isso xD

**Comment by nicholasio** [comment]
Temos que tomar cuidado com o que WordPress.org chama de "Plugin territory" só podemos colocar em metaboxes configurações que alterem o visual. O @claudiosmweb falou em usar o Customizr em vez de usar Theme Options do Odin. O que eu acho válido, pois é o caminho natural. O Theme Customizer tem painel agora e dá pra organizar bem as configurações, mesmo que sejam muitas. 

Recomendo que dêem uma olhada no Customizer.

**Comment by leobaiano** [comment]
Sim @nicholasio era a api do customizer mesmo que falaram quando tiraram o Odin. Mas vem cá, para CPT e metaboxe ela resolveria também?

**Comment by brenoalvs** [comment]
Dêem uma lida aqui, sobre o Plugin Territory.
Acho que já temos que tirar os botões de share do tema xD

https://make.wordpress.org/themes/handbook/guidelines/plugin-territory/

Não @leobaiano a customizer é só pra alterações visuais.
Dá pra criar controles, pra inserir textinhos na wp_options, ou até mesmo mudar as colunas do layout.
Mas só pra coisas visuais mesmo.

**Comment by nicholasio** [comment]
@leobaiano Algumas coisas acredito que ainda vale a pena colocar como CPT, como serviços, habilidades etc.

Além de alterações visuais no customizer dá para colocar configurações relativos ao tema: Textos estáticos, configurações de quais sections irão aparecer. Eu acho que poderíamos dar a opção para o usuário desativar algumas das sections, caso ele não queria por exemplo, colocar o portfolio.

**Comment by nicholasio** [comment]
@brenoalvs hoje em dia não é somente alterações visuais. Pois temos o Menu Customizer e o Widgets Customizer ;)

A não ser que na sua definição de alterações visuais inclua isso ;)

**Comment by brenoalvs** [comment]
Isso aí @nicholasio  o que quis dizer é que não dá pra registrar um post type por exemplo, rs.
Ou será que dá?

**Comment by leobaiano** [comment]
Então vamos copiar as classes que precisamos do Odin (CPT, metabox e taxonomy) e utilizar a Theme Customizer para as opções e definições?

**Comment by rodrigo-brito** [comment]
Massa!

**Comment by nicholasio** [comment]
@brenoalvs ah sim ;).

@leobaiano Acho que é isso mesmo. CPT, Metabox e Taxonomy. O resto é o Customizer.

**Comment by leobaiano** [comment]
Vamos dar um tempinho ai pra ver se alguém mais se pronuncia e caso não mais tarde crio as issues e já começamos a meter mão, beleza?

**Comment by brenoalvs** [comment]
Blz! 
Cc: @valeriosouza e @claudiosmweb

**Comment by claudiosanches** [comment]
Esqueçam, não vamos usar nada do Odin para isso.
Tema vai pro WordPress.org, não vamos ter framework.
Todas as opções vão no "Customizer", como o WordPress faz agora.

CPT, Metabox e taxonomy, só me dizer o que precisam.

**Comment by nicholasio** [comment]
Vamos tentar definir então o que vamos fazer com cada coisa:
Ex:
Em serviços vamos usar CPT, em "Habilidades" usar X
No customizer: Icon e texto do banner. Títulos/Subtítulos das seções etc.

**Comment by leobaiano** [comment]
Pensei aqui no seguinte:

Customize
&nbsp;&nbsp;&nbsp;&nbsp;Logo topo
&nbsp;&nbsp;&nbsp;&nbsp;Cor primaria
&nbsp;&nbsp;&nbsp;&nbsp;Cor secundaria
&nbsp;&nbsp;&nbsp;&nbsp;Dark Color
&nbsp;&nbsp;&nbsp;&nbsp;Light Gray
&nbsp;&nbsp;&nbsp;&nbsp;Redes para o share dos posts do blog
&nbsp;&nbsp;&nbsp;&nbsp;Latitude e Longitude
&nbsp;&nbsp;&nbsp;&nbsp;Telefones para contato
&nbsp;&nbsp;&nbsp;&nbsp;E-mails para contato
&nbsp;&nbsp;&nbsp;&nbsp;Endereço
&nbsp;&nbsp;&nbsp;&nbsp;Redes Sociais

CPT
&nbsp;&nbsp;&nbsp;&nbsp;Serviços
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Custom Field para definir o ícone
&nbsp;&nbsp;&nbsp;&nbsp;Habilidades
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Custom field para definir o ícone
&nbsp;&nbsp;&nbsp;&nbsp;Portfólio
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Taxonomy para categorias
&nbsp;&nbsp;&nbsp;&nbsp;Blog
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Taxonomy para categorias
&nbsp;&nbsp;&nbsp;&nbsp;Clientes

Fiquei em dúvida sobre como fazer para dar a opção do usuário definir o que aparece ou não na página inicial. Já vi alguns temas que definem uma página estatica para a home e o usuário escolhe o que aparece ou não com shortcodes, não acho um processo intuitivo e simples, porém é bem flexivel. Podemos fazer de forma menos flexivel, definindo os blocos e o usuário pode retirar um bloco ou reordenar através das opções do tema.

**Comment by rodrigo-brito** [comment]
Latitude e Longitude não precisa não... o Mapa está com localizador, da para reaproveitar o campo de endereço para fazer o marcador

**Comment by leobaiano** [comment]
Eu sei, o lance é que nem sempre bate certinho, por conta do número e a margem de erro, no caso de empresas é bom cair bem em cima do prédio, com latitude e longitude o cara pode ir no gmaps e navegar pra pegar a localização exata.

**Comment by nicholasio** [comment]
Dá pra fazer no Customizer @leobaiano escolher o que aparece ou não. Basta criar checkboxes com "exibir seção X" ou até utilizar widgets, mutos temas usam widgets para representar cada seção. E como temos agora o Widgets Customizer isso poderia ser uma boa também,  uma vez que o usuário poderia até reordenar os widgets no próprio Customizer com live preview.

Sobre o Blog podemos simplesmente usar os posts mesmo, sem precisar criar CPT e ou taxonomia.

**Comment by brenoalvs** [comment]
Essa dos widgets pras seções é interessante, só não sei como seria implementar.
Mas também dá pra fazer o live-preview e drag'n drop com esse lance das checkboxs, teríamos que ver o que é mais "correto" do ponto de vista de aprovação no repositório.

**Comment by nicholasio** [comment]
Estou em dúvida se as seções como widgets aprovaria, mas se aprovar é fácil de implementar.

Cria uma "sidebar" que na verdade seria os blocos  (sections do site) e cria um widget pra representar cada seção.

**Comment by brenoalvs** [comment]
Pois é, mas a dúvida era mais essa, criar vários widgets só pra fazer o output do HTML? Talvez tenha uma solução melhor. Mas de qualquer forma, no Customizer podemos criar um novo Controle, com uma lista de checkbox com drag'n drop e aí fazemos da forma "correta". Se é que assim pode também xD

**Comment by nicholasio** [comment]
Nunca fiz drag'n drop no customizer, se tiver como seria show de bola.

**Comment by brenoalvs** [comment]
É só usar o jquery ui, da mesma forma dos widgets.

**Comment by claudiosanches** [comment]
@leobaiano pode criar um issue com task list para isso ou issues com as coisas que precisa?
Assim já vou fechando.

**Comment by leobaiano** [comment]
Posso criar sim @claudiosmweb , vou criar um issue pra cada parada que fica mais organizado.

**Comment by leobaiano** [comment]
@claudiosmweb criei as issues, assina as que você for trabalhar. Só a do blog que ainda está em discussão se vai fazer CPT ou usar os posts padrão do WP.

**Comment by brenoalvs** [comment]
@leobaiano, fico em dúvida se realmente é necessário um CPT para os Serviços e Skills.
Eles não precisarão de URL, Archives, taxonomies, de nada xD. 

Serão usados só para exibição. Não é melhor fazer com options?
Dá pra fazer tipo uma tabelinha, ou um repeater field com ícone, título e descrição.

**Comment by nicholasio** [comment]
@brenoalvs tem razão, se não precisar de URL os CPT vão atrapalhar pois irão criar URL's sem conteúdo.

**Comment by leobaiano** [comment]
Você está certo @brenoalvs, não pensei muito quando defini essas paradas, por isso deixei aqui pra gente discutir. Vou editar as issue antes que alguém assine e comece a meter mão.

Uma coisa, options que você diz é a settings API?

**Comment by brenoalvs** [comment]
Isso! xD
Disse options no sentido de como fazemos em theme options.

**Comment by leobaiano** [comment]
sim, sim, eu saquei, só queria confirmar para poder editar a issue fazendo referência.
